core/src/main/java/io/opencensus/stats/Measure.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.common.Function;
+import io.opencensus.internal.StringUtil;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Measure.
+ */
+@Immutable
+public abstract class Measure {
+
+  public static final int MAX_LENGTH = StringUtil.MAX_LENGTH;
+
+  /**
+   * Applies the given match function to the underlying data type.
+   */
+  public abstract <T> T match(Function<DoubleMeasure, T> p0, Function<LongMeasure, T> p1);
+
+  /**
+   * Name of measure, as a {@code String}.
+   */
+  public abstract String getName();
+
+  /**
+   * Detailed description of the measure, used in documentation.
+   */
+  public abstract String getDescription();
+
+  /**
```

Comment:
Yes, here's the link to StackDriver documentation (refer to `unit` specification): https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors#resource-metricdescriptor 

Here's a complete style guide: http://unitsofmeasure.org/ucum.html